src/rcclTracker.cpp

```diff
@@ -25,12 +25,9 @@ RingNodePool_t::RingNodePool_t() {
     HIPCHECK(
         hipHostMalloc(&barrier_, sizeof(Barrier_t), hipHostMallocCoherent));
 
-    std::atomic_store_explicit(&(barrier_->bar_in), 0,
-                               std::memory_order_seq_cst);
-    std::atomic_store_explicit(&(barrier_->bar_out), 0,
-                               std::memory_order_seq_cst);
-    std::atomic_store_explicit(&(barrier_->times_done), 0,
-                               std::memory_order_seq_cst);
+    barrier_->bar_in = 0;
```

Comment:
should these use atomicExch as well?

Reply:
The HIP atomics use __device__ only, no __host__, so I can't use atomicExch here.  This is initializing the barrier values to 0 with hipHostMallocCoherent memory, and this is during rcclComm_t initialization.  Do we need to use an atomic store in this case?

Reply:
Outside of the host initializing these variables to 0, they are never again accessed by the host.  However, each GPU must be able to access these values.  I'm unsure, but I thought for atomics to work across GPUs the memory must be host mapped?

Reply:
Can we clarify across GPU atomic memory location with design team? At minimal, can we ensure consistent access to these variables by removing any access from host side, and only let GPU to set them through atomic methods?